src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -22,147 +23,162 @@ internal class ListPackageCommandRunner : IListPackageCommandRunner
+                    problemType: ProblemType.Error);
+            }
+            else
+            {
+                var lockFileFormat = new LockFileFormat();
+                LockFile assetsFile = lockFileFormat.Read(assetsPath);
 
-                // If the file was not found, print an error message and continue to next project
-                if (!File.Exists(assetsPath))
-                {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_AssetsFileNotFound,
-                        projectPath));
-                    Console.WriteLine();
-                }
-                else
+                // Assets file validation
+                if (assetsFile.PackageSpec != null &&
+                    assetsFile.Targets != null &&
+                    assetsFile.Targets.Count != 0)
                 {
-                    var lockFileFormat = new LockFileFormat();
-                    var assetsFile = lockFileFormat.Read(assetsPath);
+                    // Get all the packages that are referenced in a project
+                    IEnumerable<FrameworkPackages> packages = msBuild.GetResolvedVersions(project.FullPath, listPackageArgs.Frameworks, assetsFile, listPackageArgs.IncludeTransitive, includeProjects: listPackageArgs.ReportType == ReportType.Default);
 
-                    // Assets file validation
-                    if (assetsFile.PackageSpec != null &&
-                        assetsFile.Targets != null &&
-                        assetsFile.Targets.Count != 0)
+                    // If packages equals null, it means something wrong happened
+                    // with reading the packages and it was handled and message printed
+                    // in MSBuildAPIUtility function, but we need to move to the next project
+                    if (packages != null)
                     {
-                        // Get all the packages that are referenced in a project
-                        var packages = msBuild.GetResolvedVersions(project.FullPath, listPackageArgs.Frameworks, assetsFile, listPackageArgs.IncludeTransitive, includeProjects: listPackageArgs.ReportType == ReportType.Default);
-
-                        // If packages equals null, it means something wrong happened
-                        // with reading the packages and it was handled and message printed
-                        // in MSBuildAPIUtility function, but we need to move to the next project
-                        if (packages != null)
+                        // No packages means that no package references at all were found in the current framework
+                        if (!packages.Any())
+                        {
+                            projectModel.AddProjectInformation(message: string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoPackagesFoundForFrameworks, projectModel.ProjectName), problemType: ProblemType.Information);
```

Comment:
I think this matches Andy's comment about the abstractions. This is one of those things that can be inferred from the model itself. It doesn't have to be an info message. 

The renderer can check if there are no packages and print this.

Reply:
Seems like if we address all of these, we should be able to remove the information messages being logged from the runner and we'd only raise Errors/Warnings.

Reply:
One more useful feature from current pattern is ordering.
Without this printing following ordered output in console, we have to know exactly which error or information/warning go which and have to manually keep ordering in view. Please note current console output design made before me, now I have to add more logic into view/print in order to it doesn't break. 
While we're generating model in business logic(controller) it's very easy to reason on which order it should print, we don't have to have worry about it later in view/print, it just it prints in order it's logged. Technically here information is just another type of error/warning.

`Warning 1`
`conditional print Information1`
`Warning1`
`Information 2`
`Error 1`

Reply:
I'm not sure I understand why the ordering has to be different. 

The renderer can process the data in the same order as it gets generated.

Reply:
Current design is quite simple I just call `PrintProblems`, I don't care about internal ordering, correct ordering is already there, but without information type I have to manually check each problem and make sure they're correct place.  Please just think `Information` as another `Warning/Error` type.

https://github.com/NuGet/NuGet.Client/blob/2d98be1ae155616ca080955fe205e01574f74134/src/NuGet.Core/NuGet.CommandLine.XPlat/ListPackage/ListPackageConsoleRenderer.cs#L155-L183

Reply:
> Current design is quite simple I just call PrintProblems

IMO it doesn't separately between the model and what's displayed completely. 
For example, an information message is named a `problem`.

Reply:
Sure, removed it. Please check now.